smartfs.lua

```diff
@@ -1046,9 +1111,9 @@ smartfs.element("code", {
 
 		return self.data.code
 	end,
-	submit = function(self, fields, player)
```

Comment:
Yes, this is the one place of breaking compatibility. In the new "namespaced" way only the field named equal to the element name will be passed. Trough the last change it is possible now to pass more fields by creating the element:_getSubElement_(rel_field) method for the code element. But the fields needs to be namepaced, like mycodelement:field1 / mycodelement:field2 and it is still a hacky way

I do not have any examples for code element usage, therefore no idea how to solve it proper and if the issue really needs a solution. For my understanding it is used to wrap any other formspec code from non-smart mods. But the non-smart mods does have allways an own on_recive_fields handler so no submit is needed for this use-case.
For the second use-case to get more complex elements I prefer to redefine the new container or view element instead of code like in https://github.com/bell07/minetest-smart_inventory/blob/master/libs/smartfs-elements.lua
Do you see more use-cases?